packages/next/shared/lib/i18n/normalize-locale-path.ts

```diff
@@ -21,7 +21,10 @@ export function normalizeLocalePath(
   const pathnameParts = pathname.split('/')
 
   ;(locales || []).some((locale) => {
-    if (pathnameParts[1].toLowerCase() === locale.toLowerCase()) {
+    if (
+      pathnameParts[1] &&
```

Comment:
Are you able to reproduce this? This might cause issues elsewhere in the server if the `pathname` is being passed as an empty string here. Splitting `/` should yield `[ '', '' ]` which is handled properly here

Reply:
It only happens if `pathname = ''` which might mean the issue is earlier

Reply:
Looks like we got a test case in https://github.com/vercel/next.js/issues/31423 so added that in and fixed the original error it uncovered.